### PR TITLE
Single Select

### PR DESCRIPTION
**Pull request by kylemh**
First draft.

A lot of clean-up left in here. Lots of comments needed to explain why we sometimes style in "components" but often in the style fn.

- [x] Allow for creatable options
- [x] Add creatable story
- [x] Add async story
- [x] ~Add grouped options story~ Edit: nvm - we've disallowed this
- [x] Work with design to constrain shape of options array to avoid devs needing to create custom option components
- [x] Implement ^ constraints. `{ icon?: ReactNode; description?: string }`
- [x] Rename to SingleSelect

Sorry about the lack of tests. We'll be adding all the form tests with a Cypress playground soon!

### CODE REVIEW

File: packages/zephyr/src/Forms/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+    '&:hover': {
+      cursor: 'progress',
+    },
+  }),
+  menu: (base) => ({
+    ...base,
+    border: 'none',
+    borderTop: `1px solid ${theme.colors.pigeon200}`,
+    borderRadius: 4,
+    boxShadow: `0px 1px 4px rgba(0, 0, 0, 0.25), 0px 1px 8px rgba(0, 0, 0, 0.2)`,
```
Comment (andyhqtran) [inline packages/zephyr/src/Forms/Select.tsx:99]: Should we use polished + theme values here?

File: packages/zephyr/src/Forms/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+    color: props.isDisabled ? 'currentColor' : theme.colors.pigeon500,
+    '&:hover': {
+      color: props.isDisabled ? 'currentColor' : theme.colors.pigeon500,
+    },
+  }),
+  group: (base) => ({ ...base, paddingTop: 0, paddingBottom: 0 }),
+  groupHeading: (base) => ({ ...base, display: 'none' }),
+  indicatorSeparator: (base) => ({ ...base, display: 'none' }),
+  input: (base) => ({ ...base, backgroundColor: theme.colors.white }),
+  loadingIndicator: (base) => ({ ...base, marginRight: '-4px', color: theme.colors.pigeon300 }),
```
Comment (andyhqtran) [inline packages/zephyr/src/Forms/Select.tsx:86]: We should be able to use theme value for marginRight

File: packages/zephyr/src/Forms/Select.tsx [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+        }}
+      >
+        {description?.isHidden ? (
+          <VisuallyHidden>{description.component}</VisuallyHidden>
+        ) : (
+          description?.component
+        )}
+      </Text>
+
+      <Text
```
Comment (andyhqtran) [inline packages/zephyr/src/Forms/Select.tsx:437]: Is this suppose to be always present in the DOM?

**Review by andyhqtran — APPROVED** [review]

File: packages/zephyr/src/Forms/Select.tsx (hunk 2 above)
Comment (kylemh) [reply packages/zephyr/src/Forms/Select.tsx:86]: This isn't `tx` unfortunately

Comment (kylemh) [reply packages/zephyr/src/Forms/Select.tsx:86]: but I bet I can use `-4`

Comment (kylemh) [reply packages/zephyr/src/Forms/Select.tsx:86]: unless you mean `theme.space[-4]`?

File: packages/zephyr/src/Forms/Select.tsx (hunk 3 above)
Comment (kylemh) [reply packages/zephyr/src/Forms/Select.tsx:437]: Always rendered, but conditionally displayed.

**Review by tomaszczura — APPROVED** [review]

File: packages/zephyr/src/Forms/SingleSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+   * We want developers to be conscious of many things when dealing with asynchronously loaded options. It's not as
+   * simple as: "is it loading?""
+   */
+  loadingState?: {
+    isLoading: boolean;
+    optionsListLoadingText: string;
+  };
+
+  /**
+   * Leverages a ["CreatableSelect"](https://react-select.com/creatable) from [`react-selct`](https://react-select.com/)
```
Comment (andyhqtran) [inline packages/zephyr/src/Forms/SingleSelect.tsx:231]: Typo on react-select

**Review by andyhqtran — APPROVED** [review]

**Review by dwilt — APPROVED** [review]

**Comment by kylemh** [comment]
got approval from abby

**Pull request merged**